File: golang/uPIMulator/src/linker/kernel/instruction/cc/div_nz_cc.go
package cc

import (
	"errors"
)

type DivNzCc struct {
	condition Condition
}

func (this *DivNzCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		SZ:   true,
		SNZ:  true,
		SPL:  true,
		SMI:  true,
		TRUE: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *DivNzCc) Condition() Condition {
	return this.condition
}

File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/div_nz_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class DivNZCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(
            DivNZCC.conditions(), condition,
        )

    @staticmethod
    def conditions() -> Set[Condition]:
        return {
            Condition.SZ,
            Condition.SNZ,
            Condition.SPL,
            Condition.SMI,
            Condition.TRUE,
        }
